src/common/bustub_instance.cpp

```diff
@@ -0,0 +1,80 @@
+#include "common/bustub_instance.h"
+#include <fmt/format.h>
+#include "binder/binder.h"
+#include "catalog/table_generator.h"
+#include "recovery/log_manager.h"
+
+namespace bustub {
+
+auto BustubInstance::GetExecutionContext(Transaction *txn) -> std::unique_ptr<ExecutorContext> {
+  return std::make_unique<ExecutorContext>(txn, catalog_, buffer_pool_manager_, transaction_manager_, lock_manager_);
+}
+
+BustubInstance::BustubInstance(const std::string &db_file_name) {
+  enable_logging = false;
+
+  // storage related
+  disk_manager_ = new DiskManager(db_file_name);
+
+  // log related
+  log_manager_ = new LogManager(disk_manager_);
+
+  buffer_pool_manager_ = new BufferPoolManagerInstance(BUFFER_POOL_SIZE, disk_manager_, log_manager_);
+
+  // txn related
+  lock_manager_ = new LockManager();
+  transaction_manager_ = new TransactionManager(lock_manager_, log_manager_);
+
+  // checkpoints
+  checkpoint_manager_ = new CheckpointManager(transaction_manager_, log_manager_, buffer_pool_manager_);
+
+  // catalog
+  catalog_ = new Catalog(buffer_pool_manager_, nullptr, nullptr);
+}
+
+auto BustubInstance::ExecuteSql(const std::string &sql) -> std::vector<std::string> {
+  if (!sql.empty() && sql[0] == '\\') {
+    // internal commands
```

Comment:
```suggestion
    // Internal meta-commands, like in `psql`.
```

Since we're already borrowing PostgreSQL terminology, we might as well keep doing that.